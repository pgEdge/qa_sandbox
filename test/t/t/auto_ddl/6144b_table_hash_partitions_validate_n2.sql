--This file will run on n2 and validate all the replicated tables data, structure and replication sets they're in
SELECT pg_sleep(1);--to ensure all objects are replicated

SET ROLE appuser;

SET search_path TO s614, public;

-- Validate structure and data after adding new partition
\d sales_hash_1
\d sales_hash_2
\d sales_hash_3
\d sales_hash_4
\d sales_hash
SELECT * FROM get_table_repset_info('sales_hash'); -- Expect all partitions to be listed
SELECT * FROM sales_hash ORDER BY sale_id; -- Expect 4 rows
--exercise ddl on n2
DROP TABLE sales_hash CASCADE;

\d products_hash
\d products_hash_1
\d products_hash_2
\d products_hash_3
\d products_hash_4

SELECT * FROM get_table_repset_info('products_hash'); -- Expect the replication set to be default
SELECT * FROM products_hash ORDER BY product_id; -- Expect 4 rows
--exercise ddl on n2
DROP TABLE products_hash CASCADE;
