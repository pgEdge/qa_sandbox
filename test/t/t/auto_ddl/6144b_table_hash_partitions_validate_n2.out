--This file will run on n2 and validate all the replicated tables data, structure and replication sets they're in
SELECT pg_sleep(1);--to ensure all objects are replicated
 pg_sleep 
----------
 
(1 row)

SET ROLE appuser;
SET
SET search_path TO s614, public;
SET
-- Validate structure and data after adding new partition
\d sales_hash_1
               Table "s614.sales_hash_1"
   Column    |  Type   | Collation | Nullable | Default 
-------------+---------+-----------+----------+---------
 sale_id     | integer |           | not null | 
 sale_date   | date    |           | not null | 
 sale_amount | numeric |           |          | 
Partition of: sales_hash FOR VALUES WITH (modulus 4, remainder 0)
Indexes:
    "sales_hash_1_pkey" PRIMARY KEY, btree (sale_id, sale_date)

\d sales_hash_2
               Table "s614.sales_hash_2"
   Column    |  Type   | Collation | Nullable | Default 
-------------+---------+-----------+----------+---------
 sale_id     | integer |           | not null | 
 sale_date   | date    |           | not null | 
 sale_amount | numeric |           |          | 
Partition of: sales_hash FOR VALUES WITH (modulus 4, remainder 1)
Indexes:
    "sales_hash_2_pkey" PRIMARY KEY, btree (sale_id, sale_date)

\d sales_hash_3
               Table "s614.sales_hash_3"
   Column    |  Type   | Collation | Nullable | Default 
-------------+---------+-----------+----------+---------
 sale_id     | integer |           | not null | 
 sale_date   | date    |           | not null | 
 sale_amount | numeric |           |          | 
Partition of: sales_hash FOR VALUES WITH (modulus 4, remainder 2)
Indexes:
    "sales_hash_3_pkey" PRIMARY KEY, btree (sale_id, sale_date)

\d sales_hash_4
               Table "s614.sales_hash_4"
   Column    |  Type   | Collation | Nullable | Default 
-------------+---------+-----------+----------+---------
 sale_id     | integer |           | not null | 
 sale_date   | date    |           | not null | 
 sale_amount | numeric |           |          | 
Partition of: sales_hash FOR VALUES WITH (modulus 4, remainder 3)
Indexes:
    "sales_hash_4_pkey" PRIMARY KEY, btree (sale_id, sale_date)

\d sales_hash
          Partitioned table "s614.sales_hash"
   Column    |  Type   | Collation | Nullable | Default 
-------------+---------+-----------+----------+---------
 sale_id     | integer |           | not null | 
 sale_date   | date    |           | not null | 
 sale_amount | numeric |           |          | 
Partition key: HASH (sale_id)
Indexes:
    "sales_hash_pkey" PRIMARY KEY, btree (sale_id, sale_date)
Number of partitions: 4 (Use \d+ to list them.)

SELECT * FROM get_table_repset_info('sales_hash'); -- Expect all partitions to be listed
 nspname |   relname    | set_name 
---------+--------------+----------
 s614    | sales_hash   | default
 s614    | sales_hash_1 | default
 s614    | sales_hash_2 | default
 s614    | sales_hash_3 | default
 s614    | sales_hash_4 | default
(5 rows)

SELECT * FROM sales_hash ORDER BY sale_id; -- Expect 4 rows
 sale_id | sale_date  | sale_amount 
---------+------------+-------------
       1 | 2023-01-01 |       100.0
       2 | 2023-01-02 |       200.0
       3 | 2023-01-03 |       150.0
       4 | 2023-01-04 |       250.0
(4 rows)

--exercise ddl on n2
DROP TABLE sales_hash CASCADE;
NOTICE:  drop cascades to table sales_hash_4 membership in replication set default
NOTICE:  drop cascades to table sales_hash_3 membership in replication set default
NOTICE:  drop cascades to table sales_hash_2 membership in replication set default
NOTICE:  drop cascades to table sales_hash_1 membership in replication set default
NOTICE:  drop cascades to table sales_hash membership in replication set default
INFO:  DDL statement replicated.
DROP TABLE
\d products_hash
         Partitioned table "s614.products_hash"
    Column    |  Type   | Collation | Nullable | Default 
--------------+---------+-----------+----------+---------
 product_id   | integer |           | not null | 
 product_date | date    |           | not null | 
 product_name | text    |           |          | 
Partition key: HASH (product_id)
Indexes:
    "products_hash_pkey" PRIMARY KEY, btree (product_id, product_date)
Number of partitions: 4 (Use \d+ to list them.)

\d products_hash_1
              Table "s614.products_hash_1"
    Column    |  Type   | Collation | Nullable | Default 
--------------+---------+-----------+----------+---------
 product_id   | integer |           | not null | 
 product_date | date    |           | not null | 
 product_name | text    |           |          | 
Partition of: products_hash FOR VALUES WITH (modulus 4, remainder 0)
Indexes:
    "products_hash_1_pkey" PRIMARY KEY, btree (product_id, product_date)

\d products_hash_2
              Table "s614.products_hash_2"
    Column    |  Type   | Collation | Nullable | Default 
--------------+---------+-----------+----------+---------
 product_id   | integer |           | not null | 
 product_date | date    |           | not null | 
 product_name | text    |           |          | 
Partition of: products_hash FOR VALUES WITH (modulus 4, remainder 1)
Indexes:
    "products_hash_2_pkey" PRIMARY KEY, btree (product_id, product_date)

\d products_hash_3
              Table "s614.products_hash_3"
    Column    |  Type   | Collation | Nullable | Default 
--------------+---------+-----------+----------+---------
 product_id   | integer |           | not null | 
 product_date | date    |           | not null | 
 product_name | text    |           |          | 
Partition of: products_hash FOR VALUES WITH (modulus 4, remainder 2)
Indexes:
    "products_hash_3_pkey" PRIMARY KEY, btree (product_id, product_date)

\d products_hash_4
              Table "s614.products_hash_4"
    Column    |  Type   | Collation | Nullable | Default 
--------------+---------+-----------+----------+---------
 product_id   | integer |           | not null | 
 product_date | date    |           | not null | 
 product_name | text    |           |          | 
Partition of: products_hash FOR VALUES WITH (modulus 4, remainder 3)
Indexes:
    "products_hash_4_pkey" PRIMARY KEY, btree (product_id, product_date)

SELECT * FROM get_table_repset_info('products_hash'); -- Expect the replication set to be default
 nspname |     relname     | set_name 
---------+-----------------+----------
 s614    | products_hash   | default
 s614    | products_hash_1 | default
 s614    | products_hash_2 | default
 s614    | products_hash_3 | default
 s614    | products_hash_4 | default
(5 rows)

SELECT * FROM products_hash ORDER BY product_id; -- Expect 4 rows
 product_id | product_date | product_name 
------------+--------------+--------------
          1 | 2023-01-01   | Laptop
          2 | 2023-01-02   | Shirt
          3 | 2023-01-03   | Smartphone
          4 | 2023-01-04   | Tablet
(4 rows)

--exercise ddl on n2
DROP TABLE products_hash CASCADE;
NOTICE:  drop cascades to table products_hash_4 membership in replication set default
NOTICE:  drop cascades to table products_hash_3 membership in replication set default
NOTICE:  drop cascades to table products_hash_2 membership in replication set default
NOTICE:  drop cascades to table products_hash_1 membership in replication set default
NOTICE:  drop cascades to table products_hash membership in replication set default
INFO:  DDL statement replicated.
DROP TABLE
